### PR TITLE
pass original request as a second parameter to response/responseError

### LINKED ISSUE #33: Make request available in Interceptor.response/responseError
Could the Interceptor.response and (especially) responseError methods be extended to include the original request? This would allow us to implement 401 handling with retry as can be done with http-client using HttpResponseMessage.requestMessage.

### PR DESCRIPTION
**Pull request by timbell**
feat(interceptor): pass original request as a second parameter to
response/responseError callbacks

Pass original request to response/responseError callbacks to support,
for example, handling 401 unauthorized errors by resubmitting the
request with added credentials

Closes #33

### CODE REVIEW

File: src/http-client.js [hunk 1]
```diff
@@ -105,12 +105,13 @@ export class HttpClient {
         if (Response.prototype.isPrototypeOf(result)) {
           response = result;
         } else if (Request.prototype.isPrototypeOf(result)) {
+          request = Promise.resolve(result);
           response = fetch(result);
         } else {
           throw new Error(`An invalid result was returned by the interceptor chain. Expected a Request or Response instance, but got [${result}]`);
         }
 
-        return processResponse(response, this.interceptors);
```
Comment (bryanrsmith) [inline src/http-client.js]: The only change you should need in this method is to add `result` as a third argument to `processResponse`.

**Comment by bryanrsmith** [comment]
Thanks @timbell, looks like a good start! There are a couple things that need adjustment before I can merge.

1. Our contributing.md guidelines require a specific commit message format, which helps our release tooling to build good release notes. Something like `feat(interceptors): provide Request to response interceptors`.
2. Files in the dist directory should be reverted. We only update dist files as part of the release process.
3. Looks like the build is failing because there is some trailing whitespace that violates our lint rules. You can run these checks locally by running `gulp` in the root of the repo. (you may need to `npm install -g gulp`)

Let me know if I can provide any more info.

File: src/http-client.js (hunk 1 above)
Comment (timbell) [inline src/http-client.js]: But can't result be a Response if request or requestError return a Response directly and we'd then be passing a Response as a Request to response/responseError?

**Comment by timbell** [comment]
@bryanrsmith :

1. I did try! I wasn't sure whether the `feat(interceptors)` line should go in the body or subject - will have another go

2. Understood, will fix

3. Yes, will fix

File: src/http-client.js (hunk 1 above)
Comment (bryanrsmith) [inline src/http-client.js]: Oh yes, good point. Actually, this is probably an issue in the existing
code. I don't think it bails out of applying the request interceptors if
the first one returns a Response.

**Comment by bryanrsmith** [comment]
Looks good @timbell. Would you mind just squashing your 3 commits into one?

You can do this with an interactive rebase--
`git rebase -i 50eb3aa90bca94a8feabce09e36fdd600d813ae6`

Then choose "squash" or "fixup" for the two later commits and make sure you're happy with the commit message.

Then update the PR by force pushing to your branch `git push -f`.

**Comment by bryanrsmith** [comment]
BTW: CI build is failing, but it looks like a problem with the version of karma we're running and not a problem with these changes. I'll get that fixed this weekend.

**Comment by timbell** [comment]
@bryanrsmith, sorry I'm having difficulty with this and I don't know git very well at all.

I run:
`pick 6415626 pass original request as a second parameter to response/responseError callbacks`
`fixup 9575afd Revert "pass original request as a second parameter to response/responseError callbacks"`
`squash 4b59047 feat(interceptors): provide Request to response interceptors`

And get:

`rebase in progress; onto 50eb3aa`
`You are currently rebasing branch 'include-original-request-in-interceptor-response-callbacks' on '50eb3aa'.`

`No changes`
`You asked to amend the most recent commit, but doing so would make
it empty. You can repeat your command with --allow-empty, or you can
remove the commit entirely with "git reset HEAD^".`

`Could not apply 9575afd79976de305e2f31e73468c847f66a691e... Revert "pass original request as a second parameter to response/responseError callbacks"`

The middle commit was a revert of the first one. Can I use --allow-empty or will that muck things up? I basically want the first two commits to disappear...

**Comment by bryanrsmith** [comment]
Ah, you can just delete the lines with the two commits you don't want to keep.

**Comment by RWOverdijk** [comment]
:+1:

**Comment by alfkonee** [comment]
:+1:

**Comment by timbell** [comment]
@bryanrsmith - I've updated the PR, does it look ok now?

**Pull request merged**